error.go

```diff
@@ -0,0 +1,40 @@
+package xjson
+
+import (
+	"errors"
+)
+
+// Error implements a JSON-serializable error
+// "Err" for less confusion over func names
+type Error struct {
+	Err error
+}
+
+func NewError(e error) Error { return Error{Err: e} }
```

Comment:
Also implement `Unwrap` to support Go1.13+'s error wrapping, see https://jayconrod.com/posts/116/error-handling-guidelines-for-go and https://blog.golang.org/go1.13-errors

Reply:
@insomniacslk love these articles!